### PR TITLE
all: instrument library loading

### PR DESCRIPTION
**Pull request by TristanCrawford**
Potential fix for #1327

TL;DR This instruments the calls to `DynLib.open` to log exactly what libraries are missing and returns a consistent `LibraryNotFound` error.

image

---
As for the issue I was running into earlier: The open call in `src/sysgpu/vulkan.zig` was not catching and returning in the same manner as the other library loads, which is why I was getting that non-descript `FileNotFound` error.

One caveat is that some libraries that aren't _really_ required will still log an error, but it's definitely better than the ambiguous errors from before imo.

There may be a better name and/or a better place to put the `loadLibrary` function, so I'm open to shifting some things around if y'all see fit. Cheers!

---
- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

### CODE REVIEW

File: src/core/util.zig [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const std = @import("std");
+
+const log = std.log.scoped(.mach);
+
+// Instrumented function to load system libraries and print nicer error
+// messages.
+pub inline fn loadLibrary(libName: []const u8) !std.DynLib {
```
Comment (slimsag) [inline src/core/util.zig:7]: Can you move this [here](https://github.com/hexops/mach/blob/main/src/main.zig#L37) instead and call it `dynLibOpen`?

File: src/core/util.zig [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    return std.DynLib.open(libName) catch |err| {
+        switch (err) {
+            error.FileNotFound => {
+                log.err("Missing system library: '{s}'!\x00", .{libName});
```
Comment (slimsag) [inline src/core/util.zig:11]: not sure what the `\x00` is for?

**Comment by slimsag** [comment]
Two small comments, looks good otherwise! Thanks for improving this!

File: src/core/util.zig (hunk 1 above)
Comment (TristanCrawford) [reply src/core/util.zig]: Updated!

File: src/core/util.zig (hunk 2 above)
Comment (TristanCrawford) [reply src/core/util.zig]: Brain rot on my part. Fixed lol.

**Review by slimsag — APPROVED** [review]
